csvcubed/csvcubed/utils/qb/standardise.py

```diff
@@ -16,6 +17,39 @@
     NewQbMeasure,
     QbMultiMeasureDimension,
 )
+from csvcubed.utils.uri import uri_safe
+
+
+def standardise_categoricals(data: pd.Series) -> pd.Series:
+    """Standardise categorical data assuming case insensitivity along highest sorted() instance of uri_safe result"""
+    if data.cat.categories.dtype != "O":
+        # If it's not a string, it's not going to be a problem
+        return data
+    if len(data.cat.categories.map(lambda x: uri_safe(x)).unique()) == len(
+        data.cat.categories
+    ):
+        # There is a 1:1 relationship between categories and uri_safe categoies
+        return data
+    else:
+        # There is a *:1 relationship between categories and uri_safe categories
+        unique_keys = DefaultDict(list)
+
+        # generate all uri_safe values
+        for k, v in {value: uri_safe(value) for value in data.cat.categories}.items():
+            unique_keys[v].append(k)
+
+        cat_map = DefaultDict(str)
```

Comment:
Since you've got this functionality in two places it may be worth refactoring it out into a util. `find_uri_safe_duplicates` or something?

Reply:
Maybe with different warning messages.